src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -426,9 +426,9 @@ public boolean wouldBeChangeTo(ConsumerConfiguration serverCc) {
                 || (flowControl != null && flowControl != serverCcc.isFlowControl())
                 || (headersOnly != null && headersOnly != serverCcc.isHeadersOnly())
 
-                || CcChangeHelper.START_SEQ.wouldBeChange(startSeq, serverCcc.startSeq)
+                || CcChangeHelper.START_SEQ.wouldBeChange(startSeq, serverCcc.getStartSequence())
```

Comment:
the server sends nothing when an unsigned int is 0. By using the getter here, null is understood as 0. Not strictly necessary, but will allow users to set the value 0 and it not be considered a change if the server sends null for 0

Reply:
this has been reverted